crates/consensus/beacon/src/beacon_consensus.rs

```diff
@@ -96,3 +95,21 @@ fn validate_header_extradata(header: &Header) -> Result<(), ConsensusError> {
         Ok(())
     }
 }
+
+/// Returns true if the beacon chain consensus is active for a block with the given header and total
+/// difficulty.
+///
+/// This is the case if the block is after the Paris hardfork or the chain is configured to use
+/// optimism.
+fn is_beacon_consensus_active(
+    chain_spec: &ChainSpec,
+    header: &Header,
+    total_difficulty: U256,
+) -> bool {
+    #[cfg(feature = "optimism")]
+    if chain_spec.optimism.is_some() {
+        return true
+    }
+
+    chain_spec.fork(Hardfork::Paris).active_at_ttd(total_difficulty, header.difficulty)
```

Comment:
I'm curious why we wouldn't just configure paris block to be the genesis block and avoid a diff entirely?

Reply:
This is the plan in the `ChainSpec`s - I think this code is good to have for redundancy, but also open to not including it.

Reply:
I was aiming for consistency with op-geth, although agreed that it may make more sense to simply consider these hardforks applied at genesis in any Optimism-related chain specs. This change likely should be backported to op-geth if we decide to take this approach instead.

Reply:
Proposed the following PR as a simpler alternative, which only alters the chain spec definition: https://github.com/refcell/reth/pull/15